[Tsou Teng Yuan] iP

tsoudibi opened the pull request.

Commit:
add logo and echo

Commit:
implement list to store user input; update prefix

tsoudibi committed:
add class "task"; implement marking

tsoudibi committed:
add comment

tsoudibi committed:
fix typo

tsoudibi committed:
add deadline, todo and event

tsoudibi committed:
A-TextUiTesting

tsoudibi committed:
use RE to standerlized complex input command

Zhang-Zhitong commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,89 @@
+            "/ /___/ /_/ / /_/ / /_/ / / / /  __/    \n"+
+            "\\____/\\__,_/_.___/\\____/_/ /_/\\___/ \n";
+        String chat_prefix = "\n(Cubone) ";
+        String chat_bar = "---------------------------------";
```
you can extract this as a constant.

Zhang-Zhitong commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,89 @@
+            " / /   / / / / __ \\/ __ \\/ __ \\/ _ \\\n"+
+            "/ /___/ /_/ / /_/ / /_/ / / / /  __/    \n"+
+            "\\____/\\__,_/_.___/\\____/_/ /_/\\___/ \n";
+        String chat_prefix = "\n(Cubone) ";
```
I think we are supposed to use [camel case](https://en.wikipedia.org/wiki/Camel_case) for java and this course

Zhang-Zhitong commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,89 @@
+import java.util.regex.Pattern;
+public class Cubone {
+    public static void main(String[] args) {
+        String logo =   
```
sick lineart

Zhang-Zhitong commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,89 @@
+                // add todo task
+                String[] split = input.split(" ");
+                if(split.length == 1){
+                    System.out.println(chat_bar + chat_prefix + "☹ OOPS!!! The description of a todo cannot be empty.\n" + chat_bar);
```
this seems like a long line, also how does ☹work

Zhang-Zhitong reviewed (commented):
looks good. some nit conventions to fix

wahkia commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,89 @@
+                    System.out.println(chat_bar + chat_prefix + "Got it. I've added this Todo:\n" + inputed_tasks.get(inputed_tasks.size()-1).toString() + "\n" +
+                                        "now you have " + inputed_tasks.size() + " tasks in the list\n" + chat_bar); 
+                }
+            } else if (input.contains("deadline")){
+                // add deadline task
+                // usage: deadline <description> /by <date>
+                Pattern deadlinePattern = Pattern.compile("deadline (.+?) /by (.+)");
+                Matcher deadlineMatcher = deadlinePattern.matcher(input);
+                if (deadlineMatcher.find()) {
+                    inputed_tasks.add(new Deadline(deadlineMatcher.group(1), deadlineMatcher.group(2)));
+                    System.out.println(chat_bar + chat_prefix + "Got it. I've added this task:\n" + inputed_tasks.get(inputed_tasks.size()-1).toString() + "\n" + 
+                                        "now you have " + inputed_tasks.size() + " tasks in the list\n" + chat_bar);
```
The pattern matching and input validation for 'deadline' is slightly complicated, which could be made clearer by splitting the logic into multiple steps or using well-named variables for intermediate results.

wahkia commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,89 @@
+public class Cubone {
+    public static void main(String[] args) {
+        String logo =   
+            "   ______      __                       \n"+
+            "  / ____/_  __/ /_  ____  ____  ___     \n"+
+            " / /   / / / / __ \\/ __ \\/ __ \\/ _ \\\n"+
+            "/ /___/ /_/ / /_/ / /_/ / / / /  __/    \n"+
+            "\\____/\\__,_/_.___/\\____/_/ /_/\\___/ \n";
```
I like how you’ve included the ASCII art logo!

wahkia commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,89 @@
+                // usage: event <description> /from <date> /to <date>
+                Pattern eventPattern = Pattern.compile("event (.+?) /from (.+?) /to (.+)");
+                Matcher eventMatcher = eventPattern.matcher(input);
+                if (eventMatcher.find()) {
+                    inputed_tasks.add(new Event(eventMatcher.group(1), eventMatcher.group(2), eventMatcher.group(3)));
+                    System.out.println(chat_bar + chat_prefix + "Got it. I've added this task:\n" + inputed_tasks.get(inputed_tasks.size()-1).toString() + "\n" + 
+                                        "now you have " + inputed_tasks.size() + " tasks in the list\n" + chat_bar);
+                } else {
+                    System.out.println(chat_bar + chat_prefix + "☹ OOPS!!! Someting missing, usage: event <description> /from <date> /to <date>\n" + chat_bar);
```
There is some nesting within the conditional blocks. The nested `if` statements could be flattened by refactoring each conditional block into its own method. This would improve the readability and reduce the complexity of the code.

wahkia commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,89 @@
+            }else {
+                // add user input into list
+                inputed_tasks.add(new Task(input));
+                System.out.println(chat_bar + chat_prefix + "added: "+ input + "\n" + chat_bar);
```
I like how you used consistent UI elements like `chat_prefix` and `chat_bar` to make the user interface more engaging and clear.

wahkia reviewed (commented):
Overall, the code is well-written, but with some refactoring and attention to the guidelines, it could be made even more readable and maintainable.

tsoudibi committed:
change to switch

tsoudibi committed:
extract functions from main programs

tsoudibi committed:
add commends and exceptions

tsoudibi committed:
change the style of printing msg, avoiding long lines

tsoudibi committed:
add help

tsoudibi committed:
change exception logic

tsoudibi committed:
update eroor message

tsoudibi committed:
update comments

tsoudibi committed:
make 'Tasks' folder

tsoudibi committed:
delete unused file

tsoudibi committed:
add delete function

tsoudibi committed:
change prefix logic of tasks and add LogFile class

tsoudibi committed:
add capability of reading log file to current stage

tsoudibi committed:
Refactor reading of log file to handle invalid data

tsoudibi committed:
Merge branch 'branch-Level-7'

tsoudibi committed:
chore: ignore jar files in gitignore

tsoudibi committed:
feat: add delete function and bye command

thomasjlalba commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,319 @@
+import Tasks.Task;
+import Tasks.Todo;
+
+import utils.*;
```
Maybe it would be clearer if you import classes explicitly instead of importing everything?

thomasjlalba commented on src/main/java/utils/LogFile.java:
```diff
@@ -0,0 +1,112 @@
+                        }
+                        tasks.add(newEvent);
+                        break;
+                    default:
+                        break;
```
Maybe it would be good to add some error handling here!

thomasjlalba commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,319 @@
+            String fullCommand = sc.nextLine();
+            String[] command = fullCommand.split(" ", 2);
+            System.out.println(CHAT_BAR + CHAT_PREFIX);
+            switch (command[0]) {
```
Maybe it would be good to refactor `command[0]` to indicate what it is

thomasjlalba commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,319 @@
+        }
+    } 
+
+    /**
+     * Marks a task as done.
+     *
+     * @param index The index of the task to be marked as done.
+     */
```
Good work on implementing JavaDoc comments!

thomasjlalba commented on src/main/java/Cubone.java:
```diff
@@ -0,0 +1,319 @@
+            String[] command = fullCommand.split(" ", 2);
+            System.out.println(CHAT_BAR + CHAT_PREFIX);
+            switch (command[0]) {
+            case "bye":
```
might be good to refactor the magic strings too!

thomasjlalba commented on src/main/java/utils/LogFile.java:
```diff
@@ -0,0 +1,112 @@
+    public static File getLogFile() {
+        File file = new File(LOG_FILE_NAME);
+        // check if the file exists
+        if (!file.exists()) {
+            try {
+                // check if the dictionary exists
+                File directory = new File(file.getParent());
+                if (!directory.exists()) {
+                    directory.mkdirs();
+                }
+                file.createNewFile();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        return file;
```
Maybe you could reduce the nesting and make the happy path a bit more prominent?
```
if (file.exists()) {
  return file;
}
... // if file does not exist
```

thomasjlalba reviewed (commented):
Great work! Just some nitpicking and refactoring that could be implemented!

tsoudibi committed:
add basic ui class

tsoudibi committed:
feat: basic rewrite of command and parser

- Added new task types: [T] (To-do)
- Updated TaskList class to handle new task types
- Implemented TodoCommand to add a new to-do task
- Updated HelpCommand to include new command usage
- Removed Ui class and replaced with Messages class
- Renamed LogFile class to Storage class
- Refactored reading of log file to handle invalid data
- Added CommandResult class for command feedback
- Added Command class as base class for all commands
- Added IncorrectCommand class for handling invalid commands

tsoudibi committed:
Complete all commands

tsoudibi committed:
Refactor time formatting in deadline and event commands

tsoudibi committed:
Refactor time formatting to capable of date and time

tsoudibi committed:
feat: Add FindCommand and update HelpCommand

- Add FindCommand to search for tasks containing a keyword
- Update HelpCommand to include usage information for FindCommand

tsoudibi committed:
add missing Javadoc comments

tsoudibi committed:
Merge pull request #1 from tsoudibi/branch-Level-8

tsoudibi committed:
Merge pull request #2 from tsoudibi/branch-Level-9

tsoudibi committed:
Merge pull request #3 from tsoudibi/branch-A-JavaDoc

tsoudibi committed:
docs: Update product name and add user guide features

tsoudibi committed:
test to add theme

tsoudibi committed:
docs: Update user guide format and fix typos

tsoudibi committed:
docs: Update user guide format and fix typos

tsoudibi committed:
docs: Update task list numbering in README.md

tsoudibi committed:
dealing with exceptions

tsoudibi committed:
handle more exceptions
